Src/Newtonsoft.Json-for-Unity.Tests/Assets/SystemWebPolyfill/HttpUtility.cs

```diff
@@ -0,0 +1,670 @@
+
+            if (offset < 0 || offset > bytes.Length)
+                throw new ArgumentOutOfRangeException ("offset");
+
+            if (count < 0 || offset + count > bytes.Length)
+                throw new ArgumentOutOfRangeException ("count");
+
+            StringBuilder output = new StringBuilder ();
+            MemoryStream acc = new MemoryStream ();
+
+            int end = count + offset;
+            int xchar;
+            for (int i = offset; i < end; i++) {
+                if (bytes [i] == '%' && i + 2 < count && bytes [i + 1] != '%') {
+                    if (bytes [i + 1] == (byte) 'u' && i + 5 < end) {
+                        if (acc.Length > 0) {
+                            output.Append (GetChars (acc, e));
+                            acc.SetLength (0);
+                        }
+                        xchar = GetChar (bytes, i + 2, 4);
+                        if (xchar != -1) {
+                            output.Append ((char) xchar);
+                            i += 5;
+                            continue;
+                        }
+                    } else if ((xchar = GetChar (bytes, i + 1, 2)) != -1) {
+                        acc.WriteByte ((byte) xchar);
+                        i += 2;
+                        continue;
+                    }
+                }
+
+                if (acc.Length > 0) {
+                    output.Append (GetChars (acc, e));
+                    acc.SetLength (0);
+                }
+
+                if (bytes [i] == '+') {
+                    output.Append (' ');
+                } else {
+                    output.Append ((char) bytes [i]);
+                }
+            }
+
+            if (acc.Length > 0) {
+                output.Append (GetChars (acc, e));
+            }
+            
+            acc = null;
+            return output.ToString ();
+        }
+    
+        public static byte [] UrlDecodeToBytes (byte [] bytes)
+        {
+            if (bytes == null)
+                return null;
+
+            return UrlDecodeToBytes (bytes, 0, bytes.Length);
+        }
+
+        public static byte [] UrlDecodeToBytes (string str)
+        {
+            return UrlDecodeToBytes (str, Encoding.UTF8);
+        }
+
+        public static byte [] UrlDecodeToBytes (string str, Encoding e)
+        {
+            if (str == null)
+                return null;
+
+            if (e == null)
+                throw new ArgumentNullException ("e");
+
+            return UrlDecodeToBytes (e.GetBytes (str));
+        }
+
+        public static byte [] UrlDecodeToBytes (byte [] bytes, int offset, int count)
+        {
+            if (bytes == null)
+                return null;
+            if (count == 0)
+                return new byte [0];
+
+            int len = bytes.Length;
+            if (offset < 0 || offset >= len)
+                throw new ArgumentOutOfRangeException("offset");
+
+            if (count < 0 || offset > len - count)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/jilleJr/Newtonsoft.Json-for-Unity/pullRequest?prid=4755183)